File: solutions/scala/008-palindrome-number.scala
/**
 * Problem 8: Palindrome Number (LeetCode 9)
 * Difficulty: Easy
 * Language: Scala
 */
def isPalindrome(x: Int): Boolean = {
  if (x < 0) false
  else {
    val s = x.toString
    s == s.reverse
  }
}

File: solutions/rust/008-palindrome-number.rs
/**
 * Problem 8: Palindrome Number (LeetCode 9)
 * Difficulty: Easy
 * Language: Rust
 */
pub fn is_palindrome(x: i32) -> bool {
    if x < 0 { return false; }
    let s: Vec<char> = x.to_string().chars().collect();
    s.iter().zip(s.iter().rev()).all(|(a, b)| a == b)
}
